### ISSUE TITLE
Should services in `nixos-rebuild build-vm` work?

### ISSUE THREAD

**Issue by MichaelBergquistSuarez** [rycee/home-manager#880]
I'm using `home-manager` `release-19.03` as a nixos module.

But after `nixos-rebuild build-vm --fast && ./result/bin/run-*-vm` and logging in to my user, xmonad won't start.

I'm not yet 100% sure how to troubleshoot it, but before I continue my troubleshooting journey I thought I'd ask:

Is the `xmonad` module this is expected to work with `build-vm`?

---

I am uncertain because in [news.nix](https://github.com/rycee/home-manager/blob/761b3d0c124856433ae69c77b9c44e9b5e40e6ac/modules/misc/news.nix#L502), 2018-02-06, it says:
> For example, it is currently not possible to use 'nixos-rebuild build-vm' when using the Home Manager NixOS module

But since then this have been added to [the documentation](https://rycee.gitlab.io/home-manager/index.html#sec-install-nixos-module):
> **Note** By default packages will be installed to $HOME/.nix-profile but they can be installed to /etc/profiles if
> ```nix
> home-manager.useUserPackages = true;
> ```
> is added to the system configuration. This is necessary if, for example, you wish to use nixos-rebuild build-vm. This option may become the default value in the future.

Which I interpret as `build-vm` now is expected to work, but maybe I'm wrong? Or is it likely that there's something wrong with my setup?

# Notes
- Any tips on how I can troubleshoot this is appreciated
- `nixos-rebuild test` and `nixos-rebuild switch` works as expected, but I can't remember if I have made manual changes outside of my `configuration.nix`.
- `sudo systemctl status display-manager` shows active status.
- <details>
  <summary>Related xmonad home-manager options</summary>

  ```nix
  { config, pkgs, ... }:

  {
    home = {
      packages = [
        pkgs.haskellPackages.xmobar
      ];
    };
    xsession.windowManager.xmonad = {
      config = ./xmonad.hs;
      enable = true;
      enableContribAndExtras = true;
      extraPackages = haskellPackages: [
        haskellPackages.taffybar
        haskellPackages.xmobar 
        haskellPackages.xmonad
        haskellPackages.xmonad-contrib
        haskellPackages.xmonad-extras
      ];
    };
  }
  ```
</details>

- <details>
  <summary>Related xmonad nixos options</summary>

  ```nix
  {
    imports = [ "${home-manager}/nixos" ];
    home-manager = {
      useUserPackages = true;
      users.my-user = import ./home-manager-config;
    };
    services = {
      xserver = {
        autorun = true;
        enable = true;
        displayManager = {
          slim.enable = false;
          lightdm.enable = true;
        };
        windowManager = {
          default = "xmonad";
          xmonad.enable = true; # Each user uses home manager to configure xmonad
        };
      };
    };
    users.extraUsers.my-user = {
      description = "My user";
      isNormalUser = true;
      extraGroups = [ "wheel" ];
    }

    # I also do this in another `vm-config.nix` file used only by the vm. Could this affect somehow?
    users.extraUsers.my-user.initialPassword = "a-password";
  }
  ```
</details>

**Comment by rycee** [comment]
Yeah, it should work. Did you also set `xsession.enable = true;`? Without this the xsession would not be generated.

**Comment by MichaelBergquistSuarez** [comment]
Thank you, I did manage to get it up and running!

The suggested option did have an effect (things looked a bit better, a background image showed up) but still no xmonad

It was not a home-manager problem, so this issue can be closed, sorry for taking up time. 

I'll pay it back by explaining how I did if somebody from the future has similar problems:)
 
- `systemctl status home-manager-michael` told me that Home Manager Service failed to start with error messages such as `collect2: ld terminated with signal 9 [Killed]` and `compilation terminated`, which could indicate memory problems. 
- `dmesg` seemed to confirmed that with `Out of memory: Kill process 1374 (ld) score 409 or sacrifice child`.
- `free --mega` showed me that the VM had very little memory.

This was my first Nixos VM build so I didn't know what the defaults were or how to change them. I then found options not very well documented in [a chat](https://logs.nix.samueldr.com/nixos/2019-08-15): `virtualisation.virtualisation` and `virtualisation.memorySize`.

So I gave them a higher value, and then it simply worked, hooray
